library/core/src/array/mod.rs

```diff
@@ -364,6 +364,28 @@ impl<T: Ord, const N: usize> Ord for [T; N] {
     }
 }
 
+#[cfg(not(bootstrap))]
+#[stable(feature = "copy_clone_array_lib", since = "1.54.0")]
+impl<T: Copy, const N: usize> Copy for [T; N] {}
+
+#[cfg(not(bootstrap))]
+#[stable(feature = "copy_clone_array_lib", since = "1.54.0")]
+impl<T: Clone, const N: usize> Clone for [T; N] {
+    fn clone(&self) -> Self {
+        let mut other: [MaybeUninit<T>; N] = MaybeUninit::uninit_array();
+        for i in 0..N {
+            // SAFETY: `get_unchecked(i)` must use a valid index `i`.
+            // `self` and `other` are length N, and 0 <= i < N
+            unsafe {
+                other.get_unchecked_mut(i).write(self.get_unchecked(i).clone());
+            }
+        }
+        // SAFETY: All elements in `other` must be initialized.
+        // `other` is length N, and the prior loop iterates N times.
+        unsafe { MaybeUninit::array_assume_init(other) }
+    }
+}
```

Comment:
```suggestion

    fn clone_from(&mut self, other: &Self) {
        self.clone_from_slice(other);
    }
}
```
You could override `clone_from` here, which would use `T::clone_from` and avoid to `memcpy` the whole array.